### PR TITLE
stack/intro: Lab 8 - Stiva

### PR DESCRIPTION
**Pull request by george-cosma**
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/open-education-hub/ccas-internal/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [x] Tested your changes against relevant architectures and platforms;
- [x] Updated relevant documentation (if needed).

## Description of changes

Initial draft done, can be reviewed

### CODE REVIEW

**Comment by george-cosma** [comment]
Am sa dau fix la erori cam vineri-ish, sau pe weekend

File: chapters/stack/intro/guides/stack-addressing/README.md [hunk 1]
```diff
@@ -1,55 +1,51 @@
...
+0xff99fba0: 0xb
+0xff99fb9c: 0xc
+0xff99fb98: 0xd
+```
 
-For covenince, here's the contents of the file. To play around with it, download the lab locally.
+> **Note:** The last 4 values are the ones we pushed on stack.
+> What is the first one?
+
+> **Answer:** It is the old EBP we push at the start of the function.
```
Comment (george-cosma) [inline chapters/stack/intro/guides/stack-addressing/README.md:16]: De ce old_ebp sub noi? E pusa de catre _start, sigur, dar gen -- de ce acolo? Acolo se termina stiva?

Comment (teodutu) [reply chapters/stack/intro/guides/stack-addressing/README.md:16]: Use English for everything so we stay consistent.

As for the explanation, the `main()` function from `stack-addressing.asm` pushes 5 values onto the stack:

```assembly
push ebp
mov ebp, esp

push dword 10
push dword 11
push dword 12
push dword 13
```
The first two instructions are there to set up `main()`'s stack frame.

The first value that is pushed is the old `ebp`. As for who sets it, it's more complicated. It's not that `_start` calls `main()`, but that `_start` calls `__libc_start_main()` which initialises the process's VAS and only then calls `main()`. You can find a more detailed explanation here [1]. So the value of the old `ebp` is the beginning of `__libc_start_main()`'s stack.

[1] http://dbp-consulting.com/tutorials/debugging/linuxProgramStartup.html

File: chapters/stack/intro/guides/stack-addressing/support/stack-addressing.asm [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    push dword 11
+    push dword 12
+    push dword 13
+
+    mov eax, ebp
+print_stack:
+    PRINTF32 `0x\x0`
+    PRINTF32 `%x\x0`, eax
+    PRINTF32 `: 0x\x0`
+    PRINTF32 `%x\n\x0`, [eax]
```
Comment (teodutu) [inline chapters/stack/intro/guides/stack-addressing/support/stack-addressing.asm:21]:
```suggestion
    PRINTF32 `%p: %p\n\x0`, eax, [eax]
```
There's no point in splitting up those prints.

File: chapters/stack/intro/media/the-stack-growth.drawio.svg
Comment (teodutu) [inline chapters/stack/intro/media/the-stack-growth.drawio.svg:1]: Rename the file to `the-stack-growth.svg`.

File: chapters/stack/intro/media/the-stack.drawio.svg
Comment (teodutu) [inline chapters/stack/intro/media/the-stack.drawio.svg:1]: Ditto: `the-stack.svg`.

**Comment by george-cosma** [comment]
Don't know how to fix spellcheck. Otherwise done 👍

File: chapters/stack/intro/reading/README.md
Comment (teodutu) [inline chapters/stack/intro/reading/README.md:1]: Split this file into the following files and group the exercises and demos around them (where possible) so they form subchapters:
- `introduction.md` (intro + stack data structure)
- `stack-operations.md`
- `stack-addressing.md`

File: config.yaml [hunk 3]
```diff
@@ -233,6 +233,29 @@ docusaurus:
...
+                    path: drills/
+                    subsections:
+                      - Tasks:
+                          path: tasks/
+                          subsections:
+                            - Max/: max/
+                            - Reverse Array/: reverse-array/
+                            - Stack Addressing/: stack-addressing/
+                            - Local Var/: local-var/
+                            - Bonus - GCD/: gcd/
```
Comment (teodutu) [inline config.yaml:258]: After you split the reading material into subchapters, rearrange the guides and tasks and associate them with the newly created subchapters.

File: chapters/stack/intro/media/the-stack-growth.svg
Comment (teodutu) [inline chapters/stack/intro/media/the-stack-growth.svg:1]: Nice images!

**Review by teodutu — CHANGES_REQUESTED** [review]
I made inline comments with what needs changing. To fix spellcheck, add the false positives to these wordlists [1] and keep them sorted and avoid duplicates using `sort -u -o <file-name> <file-name>`.

[1] https://github.com/open-education-hub/actions/tree/main/spellcheck/config

File: chapters/stack/intro/reading/README.md
Comment (george-cosma) [reply chapters/stack/intro/reading/README.md:1]: The content is quite short, I think splitting up the concepts will just end up with more confused students. Teach them what they need to know, then let them practice. Splitting it off so atomically is just not worth it imho

Comment (teodutu) [reply chapters/stack/intro/reading/README.md:1]: Ok then remove the "Introduction" subchapter in `config.yaml` and make the links in the tasks refer to specific sections in the reading material (using `#`, such as `.../README.md#stack-operations`).

**Pull request merged**